tests/Argentina/ArgentinaTest.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+declare(strict_types=1);
+/*
+ * This file is part of the Yasumi package.
+ *
+ * Copyright (c) 2015 - 2021 AzuyaLabs
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @author Sacha Telgenhof <me@sachatelgenhof.com>
+ */
+
+namespace Yasumi\tests\Argentina;
+
+use ReflectionException;
+use Yasumi\Holiday;
+use Yasumi\tests\ProviderTestCase;
+
+/**
+ * Class for testing holidays in Argentina.
+ */
+class ArgentinaTest extends ArgentinaBaseTestCase implements ProviderTestCase
+{
+    /**
+     * @var int year random year number used for all tests in this Test Case
+     */
+    protected $year;
+
+    /**
+     * Initial setup of this Test Case.
+     */
+    protected function setUp(): void
+    {
+        $this->year = $this->generateRandomYear(1980);
+    }
+
+    /**
+     * Tests if all official holidays in Argentina are defined by the provider class.
+     *
+     * @throws ReflectionException
+     */
+    public function testOfficialHolidays(): void
+    {
+        $this->assertDefinedHolidays([
```

Comment:
This asserts that these holidays all are defined from the start test year (1980), which is correct. However it doesn't assert that the respective holidays are not present before their establishment year.

Reply:
But each test file has a method called `testNotHoliday`, which does exactly that (I know that they run for sure because they failed and I had to fix them), this structure of testing I copied from other providers so I think for consistency we should keep it as is.

But if you think otherwise, please let me know how you would like me to change the structure of the files?

Reply:
You're correct; I confused with some other edge cases I worked on a while ago (Sometimes I don't even remember my own code ;) )

Reply:
Just ran your test again and it fails:
```
1) Yasumi\tests\Argentina\ArgentinaTest::testObservedHolidays
Failed asserting that an array has the key 'carnavalMonday'.
```
You can replicate this by setting the starting test year lower than the establishment year. The error might not advertise itself as the actual test year is randomized.

Reply:
@stelgenhof I'm not sure what are you asking me to do here.

Each test has its own `ESTABLISHMENT_YEAR` on which the whole test is based, this is what I found in the other providers.

> you can replicate this by setting the starting test year lower than the establishment year.

Yes, but it's set to a specific year on `setUp`, it will not fail unless you change the year.
The tests will fail if you change anything in the code, in this state, it will not fail.

<img width="920" alt="Screen Shot 2021-11-24 at 9 13 22" src="https://user-images.githubusercontent.com/7369740/143191929-582a29ef-a4e1-44e1-b63a-48bf88889ea0.png">
<img width="932" alt="Screen Shot 2021-11-24 at 9 13 39" src="https://user-images.githubusercontent.com/7369740/143191938-57ecc4fd-d128-4c52-bb41-96fb35261cb4.png">



In any case, please let me know exactly what you want me to do if you think I'm wrong, I'm not sure what I need to do.

Reply:
See for example this: https://github.com/azuyalabs/yasumi/blob/develop/tests/Georgia/GeorgiaTest.php#L65..L67

Your tests will not fail as your initial test start year (1980) is after any of the establishment years. It possibly can fail if you instantiate Yasumi with a year *before* any of the establishment years. Agreeably users might not do that a lot, but still is possible in case you'd like to see a backdated list of holidays.